src/test/storage/constraints_test.cpp

```diff
@@ -0,0 +1,93 @@
+#include <memory>
+
+#include "base_test.hpp"
+
+#include "concurrency/transaction_context.hpp"
+#include "concurrency/transaction_manager.hpp"
+#include "expression/expression_functional.hpp"
+#include "expression/pqp_column_expression.hpp"
+#include "hyrise.hpp"
+#include "operators/delete.hpp"
+#include "operators/get_table.hpp"
+#include "operators/insert.hpp"
+#include "operators/projection.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/update.hpp"
+#include "operators/validate.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+class ConstraintsTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    {
+      TableColumnDefinitions column_definitions;
+      column_definitions.emplace_back("column0", DataType::Int, false);
+      column_definitions.emplace_back("column1", DataType::Int, false);
+      column_definitions.emplace_back("column2", DataType::Int, false);
+      column_definitions.emplace_back("column3", DataType::Int, false);
+      auto table = std::make_shared<Table>(column_definitions, TableType::Data, 2, UseMvcc::Yes);
+
+      auto& sm = Hyrise::get().storage_manager;
+      sm.add_table("table", table);
+
+      table->add_soft_unique_constraint({ColumnID{0}}, false);
+    }
+
+    {
+      TableColumnDefinitions column_definitions;
+      column_definitions.emplace_back("column0", DataType::Int, false);
+      column_definitions.emplace_back("column1", DataType::Int, true);
+      auto table = std::make_shared<Table>(column_definitions, TableType::Data, 2, UseMvcc::Yes);
+
+      auto& sm = Hyrise::get().storage_manager;
+      sm.add_table("table_nullable", table);
+
+      table->add_soft_unique_constraint({ColumnID{0}}, false);
```

Comment:
Is this needed in the tests?

Reply:
Yes, for example in the exception case that you quoted above.